android/modules/locale/src/java/ti/modules/titanium/locale/LocaleModule.java

```diff
@@ -33,6 +39,62 @@ public LocaleModule()
 		super("Locale");
 	}
 
+	@Kroll.setProperty
+	public void setApplicationLocales(String locales)
+	{
+		if (Build.VERSION.SDK_INT < 33) {
+			Log.w(TAG, "This property is only supported on Android API level 33 and above.");
+			return;
+		}
+		LocaleListCompat appLocale = LocaleListCompat.forLanguageTags(locales);
+		AppCompatDelegate.setApplicationLocales(appLocale);
+	}
+
+	@Kroll.getProperty
+	public KrollDict[] getApplicationLocales()
+	{
+		if (Build.VERSION.SDK_INT < 33) {
+			Log.w(TAG, "This property is only supported on Android API level 33 and above.");
+			return new KrollDict[0];
+		}
+		LocaleListCompat localeListCompat = AppCompatDelegate.getApplicationLocales();
+		int size = localeListCompat.size();
+		KrollDict[] locales = new KrollDict[size];
+		for (int i = 0; i < size; i++) {
```

Comment:
The loop looks a bit "oldskool". Things to change:
- Use a more modern loop technique, e.g. [this one](https://stackoverflow.com/a/18410489/5537752) and append it a to an array instead of maintaining an `index`
- Do not add all the logic in the null check. Instead, check for `locale == null` and `continue` early